src/drivers/qshell/qurt/qshell.cpp

```diff
@@ -155,11 +157,16 @@ int QShell::run_cmd(const std::vector<std::string> &appargs)
 	//replaces app.find with iterator code to avoid null pointer exception
 	for (map<string, px4_main_t>::iterator it = apps.begin(); it != apps.end(); ++it) {
 		if (it->first == command) {
-			const char *arg[2 + 1];
+			// one for command name, one for null terminator
+			const char *arg[MAX_ARGS + 2];
 
 			unsigned int i = 0;
 
-			while (i < appargs.size() && appargs[i].c_str()[0] != '\0') {
+			if (appargs.size() > MAX_ARGS + 1) {
+				PX4_ERR("%d too many arguments in run_cmd", appargs.size() - (MAX_ARGS + 1));
+			}
+
+			while (i < appargs.size() && appargs[i].c_str()[0] != '\0' && i < MAX_ARGS + 1) {
```

Comment:
`&& i < MAX_ARGS + 1` is then not needed anymore, since you already have `if (appargs.size() > MAX_ARGS + 1) {`